### PR TITLE
Add user statistics service interface

### PR DESCRIPTION
**Pull request by FredrikWallstrom**
This is related with:
https://github.com/Lundalogik/lime-webclient/pull/1266

### CODE REVIEW

File: packages/interfaces/src/service/user-statistics-service.interface.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Service for tracking user statistics to `Segment`: https://segment.com/
+ *
+ *
+ * Which event is the webclient using? move the enum events to here instead
```
[the remark this answers is not in the document]
Comment (FredrikWallstrom) [reply packages/interfaces/src/service/user-statistics-service.interface.ts:5]: Ops. These comments was only a reminder for me, forgot to remove them.

File: packages/interfaces/src/service/user-statistics-service.interface.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    sendTrackEvent(trackName: TrackEventType | string, properties?: any);
+}
+
+/**
+ * These are the track events that are being used by default in the web client.
+ */
+export enum TrackEventType {
+    /**
+     * Tracks when a user clicks on the global search.
```
[the remark this answers is not in the document]
Comment (FredrikWallstrom) [reply packages/interfaces/src/service/user-statistics-service.interface.ts:35]: Good point!

File: doc/tutorials/hello-crm.mdx [hunk 3]
```diff
@@ -128,3 +128,28 @@ private async handleClick() {
 ```
 
 Try it out and press the button. The name of the company is updated automatically!
+
+## Track with User Statistics Service
+
+Lets track to `Segment`, https://segment.com, when the name of the company has been changed.
+To track this we are going to use the `UserStatistics` service from the platform.
+
+The `handleClick` button from previous example together with a tracking event to `Segment`
```
[the remark this answers is not in the document]
Comment (FredrikWallstrom) [reply doc/tutorials/hello-crm.mdx:137]: I fix!

**Comment by FredrikWallstrom** [comment]
ping @BregenzerK

**Comment by FredrikWallstrom** [comment]
I think the reason for mention Segment in the docs was to actually make the consumer aware of where logs are stored. This because we are using a payment plan with Segment and I was thinking that we don't want to start to log wild and crazy everywhere.
If consumer don't know where the logs are being stored, why would they use it?

I agree with you regarding the `TrackEventType` enum, the reason to expose it was because consumer maybe wants to track same events that are being used by the webclient.

**Comment by BregenzerK** [comment]
I see your point with Segment, you're right that there is no good reason to mention them in the docs. But I think there should be a difference between logging and tracking statistics, don't everything that you might want to log you need statistics on. and as Fredrik said, the idea with `TrackEventType` is to collect the same dataset from the webclient as from an addon, e.g. the `TABLE_VIEW_CLICKED` is as relevant from InfoTiles than from core features and in the end you want statistics per customer not per addon.

**Comment by jgroth** [comment]
I still don't see a big difference in logging and tracking :smile: As I see it tracking what a user does is logging, and right now we store those logs in Segment. Why should we use a different way of logging depending on where we want to store it? :slightly_smiling_face: In my suggestion, it would be up to the logging handler to decide what is logged where, i.e.

```typescript
logger.info('This is not sent to Segment');
logger.info('TRACK_IN_SEGMENT', {track: {data: 'custom data'}});
```

In this example, we could have the first one just being sent to the console, but the second one is sent to Segment. The consumer still use the same logging service. How we decide where to log stuff is up for discussion, but the segment logger could decide just to log stuff it the `track` property is set on the context for instance.

Regarding the `TABLE_VIEW_CLICKED` example above, I think we have a problem if an addon has to remember to do this kind of logging on it's own. It would be very easy to "cheat" the system and not do the logging. It should be up to the implementation of the feature we want to track to do the logging. But  maybe I'm not seeing the whole picture here. Can you give an example of what you want to do and why it was not working before :slightly_smiling_face:

**Comment by FredrikWallstrom** [comment]
Ok I follow what you mean, and it seems legit for me.

What we want to do in our add-ons is to is to have a way to know which customers that actually using which add-ons. The thought here as a first step was to add this tracking in all of our add-ons, for example when the ACD-dialog has been opened we want to track that in some way. So yes, the idea was to to this logging by our own in the add-ons.
And in future considerations we maybe want to track on interaction with an add-on, for example when a tile is clicked in the info-tiles add-on. However I'm not sure if the plan was to use the same trackeventnames as the web client, I think the plan was to come up with own track names.

The second step in this process is to actually use this events in Sublime and create different product cards and so on, but this is out of scope in this PR.

**Comment by jgroth** [comment]
Ah I see :slightly_smiling_face: Then I would say that it should be up to the platform/webclient to do this kind of logging when it loads the addon, otherwise it would force addon authors to remember to do it on their own.

**Comment by FredrikWallstrom** [comment]
I believe that's OK for me.
I have been requested to leave this at the moment and focus on other stuff.

But to wrap it up. Revert this PR and then make a feature request in webclient?

**Comment by jgroth** [comment]
Sounds good :+1: 
What do you think @BregenzerK ? You have been working more with the user statistics than me :slightly_smiling_face:

**Comment by BregenzerK** [comment]
I've seen it from the end result that you want to in segment in an ideal world and from what I remember there are 2 scenarios 
- filter all tracking events from a user to observe the path through the application
- filter on one event type to see how a feature is used 

the tracks we have in the webclient are just a start and for a more complete picture we should definitely include events from addons but it will be hard to maintain a common tracking strategy in any case and I'm not sure mixing it with logging would help with that.

**Comment by FredrikWallstrom** [comment]
Yes true, we probably need a way to track more stuff within the add-ons.

**Pull request merged**